doc/iterations/2024.5/shapes/configurable_visibility.adoc

```diff
@@ -0,0 +1,47 @@
+= Configurable visibility
+
+== Problem
+
+Sirius Web doesn't provide a way for specifiers to set the default visibility of a node (hidden/visible). 
+It also doesn't provide a way to let specifiers define tools that hide/reveal nodes as part of other operations (e.g. to reveal a hidden node when a child node is created).
+Nodes are always created visible, and the only way to hide them right now is that the end user does it manually.
+
+== Key Result
+
+- Specifiers must be able to define NodeDescriptions that are hidden/visible by default.
+- Specifiers must be able to call hide/reveal services in their tool to hide/reveal a set of nodes.
+
+== Solution
+
+A new field _Is Hidden By Default Expression_ should be displayed in NodeDescription. 
+The field contains an boolean expression which is evaluated on the first rendering of the node to set its initial hidden/visible state.
+If the expression is empty it is evaluated to `false`, which means that the node is visible by default (this preserves Sirius Web's default behavior to show new nodes by default). 
+
+The `diagramServices` variable should contain new services to hide/reveal nodes.
+
+== Scenario
+
+1. Default Visibility State
+- A specifier adds a NodeDescription in the view DSL
+- The _Is Hidden By Default Expression_ field is available and its default value is empty, which means visible
+- The specifier can define an AQL expression to that returns a boolean value to set the default visibility of the node (`true` for hidden, `false` for visible)
+- When an end user creates a node represented in this manner in a diagram, the node is hidden/visible according to this expression
+- If the user manually hides/reveals the node the value returned by the expression is not taken into account (the expression can't override user-defined state).
+
+2. Hide/Reveal operation
+- A specifier creates a tool in the view DSL
+- The specifier creates an operation, for example ChangeContext
+- The specifier fills the expression of its operation with a call to the hide/reveal service, for example `diagramServices.hide(nodeId)`
+- When an end user uses the tool defined this way, the appropriate nodes are hidden/revealed
+
+=== Cutting backs
+
+This shape doesn't define any support for faded elements.
```

Comment:
And this is the part where I will disagree since it does not bring any additional complexity at all :)

Reply:
removed